### PR TITLE
Rt 16.3 policy rules

### PR DESCRIPTION
**Pull request by rameshtathe**
I have made changes, Please review
1) Updated csvs_converter.py
2) Updated monitor_converter.py
3) Updated netscaler_parser.py
4) Updated netscaler_converter.py

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -59,3 +59,4 @@ python/avi/sdk/utils/f5_converter/output/
 python/avi/f5_converter/output/
 python/avi/netscaler_converter/output/
 python/avi/netscaler_converter/test/test_output/
+.idea/
```
Comment (grastogi23) [inline .gitignore:62]: is this needed?

File: python/avi/netscaler_converter/csvs_converter.py [hunk 2]
```diff
@@ -154,3 +182,145 @@ def get_target_vs_from_policy(self, policy_lables, name, lbvs_bindings):
...
+            matches = re.findall('[0-9]+.[[0-9]+.[0-9]+.[0-9]+', query)
+            for match in matches:
+                policy_obj["http_request_policy"]["rules"][0]["match"]["client_ip"]["addrs"].append({"type": 'V4',"addr": match})
+
+        elif 'HTTP.REQ.HEADER' in ns_rule and \
+                        ".CONTAINS" in ns_rule:
+            policy_obj["http_request_policy"]["rules"][0]["match"].update({"hdrs": header})
+            policy_obj["http_request_policy"]["rules"][0]["match"]["hdrs"][0]["match_criteria"] = "HDR_EQUALS"
+
+            matches = re.findall('\"(.+?)\"', query)
```
Comment (grastogi23) [inline python/avi/netscaler_converter/csvs_converter.py:273]: what if there are no matches?

File: python/avi/netscaler_converter/csvs_converter.py [hunk 3]
```diff
@@ -154,3 +182,145 @@ def get_target_vs_from_policy(self, policy_lables, name, lbvs_bindings):
...
+                    'match': {},
+                    'switching_action': {
+                        'action': 'HTTP_SWITCHING_SELECT_POOL',
+                        'status_code': 200,
+                        'pool_ref': policy['targetLBVserver'] + '-pool'
+                    }
+                }]
+            },
+        }
+        query = ns_rule.strip('"')
```
Comment (grastogi23) [inline python/avi/netscaler_converter/csvs_converter.py:233]: might be good to also remove the blank spaces.

File: python/avi/netscaler_converter/csvs_converter.py [hunk 4]
```diff
@@ -154,3 +182,145 @@ def get_target_vs_from_policy(self, policy_lables, name, lbvs_bindings):
...
+        if 'URL ==' in ns_rule:
+            a, b = query.split("==")
+            b = b.strip()
+            match_str = b.strip("\\'")
+            policy_obj["http_request_policy"]["rules"][0]["match"].update({"path": path_query})
```
Comment (grastogi23) [inline python/avi/netscaler_converter/csvs_converter.py:238]: can there not be a case that there are multiple rules in a single vs. this code may not work then?

**Review by grastogi23 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (chaitanyaavi) [reply .gitignore:62]: Yes. Its an pycharm files. We dont need to commit it.

**Pull request merged**